src/gcp/cloudfunctions.js

```diff
@@ -20,7 +20,10 @@ function _functionsOpLogReject(func, type, err) {
   } else {
     logger.info(err.message);
   }
-  return Promise.reject(err.message);
+  return Promise.reject({
```

Comment:
Hmm. I'm still not 100% behind this. Maybe @mbleigh has a better opinion on how this should work.

Personally, I don't think we should be returning objects like this in `reject`. This also effects every function that is using this helper, so I reasonably say that this isn't breaking anything. Previously returning a string is an interesting choice, so I'm not 100% sure what the author's intention was. I would rather see a `FirebaseError` being rejected with `{ original: err, context: func }` which can be referenced in other code where appropriate.

That said - I'll ping @laurenzlong and/or @mbleigh on this. They're the original authors of this codepath, so I'd rather have them comment on this change.

Reply:
Promise rejections should always be an `Error`. Are we actually using the rejected value anywhere? If not, just `reject` with `err`. If so, you could either decorate the error with the function (`err.context = {function: func}` or something) or create a new error that wraps the original.

But yeah, in general, Promise rejections should always be `Error` objects. The original code here actually looks wrong since it rejects with `err.message` instead of `err`.

Reply:
We're not using the rejected value anywhere. I'll change it to adding to the context of the `err` as suggested.